doc/semantic_version.md

```diff
@@ -0,0 +1,43 @@
+# WAMR uses semantic versioning
+
+WAMR uses the _semantic versioning_ to replace current _date versioning_ system.
+
+There are three parts in the new version string:
+
+- _major_. WAMR only changes _major_ when APIs are modified in an incompatible way.
+  APIs includes: `wasm_export.h`, `wasm_c_api.h`, _sections in AOT files_, and so on.
+- _minor_. It represents new features. It includes not just MVP or POST-MVP features
+  but also WASI features and WAMR private ones.
+- _patch_. It represents patches.
+
+## Legacy versions
+
+All legacy versions(tags) will keep their current status. We treat the first releasing,
+_WAMR-11-28-2019_, as the version 1.0.0, and so on.
```

Comment:
what's the point to assign semver to old releases?

Reply:
Just in case somebody is interested in why our next release would be `1.20.0`

Reply:
FYI, we won't change any existing release names or links.